tests/highlevel.rs

```diff
@@ -676,3 +677,68 @@ async fn sftp_test_from_sessions() {
         sftp.close().await.unwrap();
     }
 }
+
+#[tokio::test]
+/// Test write buffer limit for tokio compat file
+async fn sftp_tokio_compact_file_write_buffer_limit() {
+    let path = gen_path("sftp_tokio_compact_file_write_buffer_limit");
+    let content = b"HELLO, WORLD!\n".repeat(100);
+    let option = SftpOptions::new()
+        .flush_interval(Duration::from_millis(100))
+        .tokio_compat_file_write_limit(NonZeroUsize::new(1000).unwrap());
+
+    let (mut child, sftp) = connect(option).await;
+    let (mut child2, sftp2) = connect(Default::default()).await;
+
+    let content = &content[..min(sftp.max_write_len() as usize, content.len())];
+    assert!(content.len() > 1000 && content.len() < 2000);
+
+    let read_entire_file = || async {
+        let mut buffer = Vec::with_capacity(content.len());
+
+        let file = sftp2
+            .open(&path)
+            .await
+            .map(file::TokioCompatFile::from)
+            .unwrap();
+        tokio::pin!(file);
+        file.read_to_end(&mut buffer).await.unwrap();
+
+        buffer
+    };
+
+    {
+        let mut fs = sftp.fs();
+
+        let file = sftp
+            .options()
+            .write(true)
+            .create_new(true)
+            .open(&path)
+            .await
+            .map(file::TokioCompatFile::from)
+            .unwrap();
+        tokio::pin!(file);
+
+        let len = content.len();
+
+        let (content1, content2) = content.split_at(len / 2);
+
+        // Create new file (fail if already exists) and write to it.
+        file.write_all(content1).await.unwrap();
+        debug_assert_eq!(read_entire_file().await.len(), 0);
+        file.write_all(content2).await.unwrap();
+        debug_assert_eq!(read_entire_file().await.len(), content1.len());
```

Comment:
There's no guarantee that `content2` won't be flushed in the meantime by the `flush_task`.

Reply:
Yes. `content1` and `content2` are both possible to be flushed. I tested some `flush_interval` options, and 100 milliseconds option is usually available here.
https://github.com/openssh-rust/openssh-sftp-client/blob/b272447a90cdad3629f7031cfa6aec82bfc3f336/tests/highlevel.rs#L687

Reply:
I am worried about this since it is quite fragile.

I would recommend setting `flush_interval` to `Duration::from_seconds(u64::MAX)`, then manually trigger flushing before writing `content2`, then wait for a few seconds and then write `content2`.

You can add a new `#[cfg(feature = "__ci-tests")]` to `Sftp` that forces `flush_task` to flush, by calling `Auxiliary::trigger_flushing`.

Reply:
@silver-ymz Turns out that you can just set `flush_interval` to some really large value since `poll_flush` will call `trigger_flushing` and my own proposal also takes care of this.

Then you would need to set `max_pending_requests` to `1` and then drop `File` before removing the file to make sure it will get flushed.